Changes to TYPE_OF_LOCATORS mapping & addition of custom Exceptions

ajanderson1 opened the pull request:
As per the commit descriptions, I propose changes to the locator mapping dictionary: `TYPE_OF_LOCATORS` in order to align with the definitions in https://github.com/SeleniumHQ/selenium/blob/trunk/py/selenium/webdriver/common/by.py

I would expect this may be a breaking change - In fact, by leaving in prior dict keys it maybe needn't be.

Additionally, added custom exceptions to expand the usability of the on-the-fly WebElement attributes, allowing us to handle a custom exception rather than excepting *all* Exceptions arising during this execution.

Would very much appreciate comment/feedback if you agree or would like discuss.

Thanks

ajanderson1 committed:
Update Pagefactory.py

changethe `TYPES_OF_LOCATORS` mapping dict in order to match the definitions in https://github.com/SeleniumHQ/selenium/blob/trunk/py/selenium/webdriver/common/by.py

This allows for direct re-use of the locators by selenium native functions where PageFactory does not offer required functionality.

Workaround: at present I have used the following workaround to 'convert' locators for use by selenium:
```python
    locators_el_loc = {key: (PageFactory.TYPE_OF_LOCATORS[value[0].lower()], value[1]) for key, value in locators.items()}
```
which allows for following usage:
```python
my_driver.find_element(*locators_el_loc['my_locator_name'])
```

ajanderson1 committed:
Update Pagefactory.py

Changed Exception to a custom PageFactory Exception in order to allow for try/except only the element not found Exception rather than the many other errors, that could potentially arise during.

NayakwadiS commented:
@ajanderson1 I appreciate your efforts in creating custom exceptions. Good Job.

but proposed change of `TYPE_OF_LOCATORS` needn't be required as per standard practice dict keys shouldn't have space and this is not required to be aligned with Selenium doc.

Kindly revert this, remaining we can merge.

Thanks.

ajanderson1 committed:
Update Pagefactory.py

Reverted `TYPE_OF_LOCATORS` as requested: https://github.com/NayakwadiS/selenium-page-factory/pull/15#issuecomment-1470190894

ajanderson1 commented:
Reverted `TYPE_OF_LOCATORS ` as requested.  With that however,  I'd push the point further :)...

When using `PageFactory`, I found there were instances where I still required the use the locators in a format that selenium could understand (ie where PageFactory methods weren't sufficient to achieve goals) and as a result required the 'translation' (below), in order to use the same locator definition.
```python
    locators_el_loc = {key: (PageFactory.TYPE_OF_LOCATORS[value[0].lower()], value[1]) for key, value in locators.items()}
```
A small point but worth bending 'best practise' for in my view.

I defer to your judgement however, and would like to add that I have very much enjoyed using + learnt a great deal from getting my head around this package. Thanks again!

NayakwadiS commented:
Change available in next release
